mkdocs/config/config_options.py

```diff
@@ -199,6 +200,58 @@ def run_validation(self, value: object) -> list[T]:
         return [fake_config[k] for k in fake_keys]
 
 
+class DictOfItems(Generic[T], BaseConfigOption[Dict[str, T]]):
+    """
+    Validates a dict of items. Keys are always strings.
```

Comment:
Hmm it doesn't actually validate that the keys are strings. Could you add that, and a test?

Reply:
Something like this? :thinking: 

```python
for key_name in fake_config:
    Type(str).run_validation(key_name)
```

Reply:
Definitely isinstance

Reply:
Done :slightly_smiling_face: